packages/dds/tree/src/simple-tree/api/treeApiBeta.ts

```diff
@@ -91,11 +103,61 @@ export const TreeBeta = {
 	 * @returns A callback function which will deregister the event.
 	 * This callback should be called only once.
 	 */
+	on<K extends keyof TreeChangeEventsBeta<TNode>, TNode extends TreeNode>(
+		node: TNode,
+		eventName: K,
+		listener: NoInfer<TreeChangeEventsBeta<TNode>[K]>,
+	): () => void;
+
+	/**
+	 * Clones the persisted data associated with a node. Some key things to note:
+	 * - Local state, such as properties added to customized schema classes, will not be cloned. However, they will be
+	 * initialized to the state after running the constructor, just like if a remote client had inserted the same node.
+	 * - Primitive node types (i.e., numbers, strings, booleans, nulls and fluid handles) will be returned as is.
+	 * - The identifiers in the node's subtree will be preserved, i.e., they are not replaced with new values.
+	 *
+	 * @param node - The node to clone.
+	 * @returns A new unhydrated node with the same persisted data as the original node.
+	 */
+	clone<TSchema extends ImplicitFieldSchema>(
+		node: TreeFieldFromImplicitField<TSchema>,
+	): TreeFieldFromImplicitField<TSchema>;
+} = {
 	on<K extends keyof TreeChangeEventsBeta<TNode>, TNode extends TreeNode>(
 		node: TNode,
 		eventName: K,
 		listener: NoInfer<TreeChangeEventsBeta<TNode>[K]>,
 	): () => void {
 		return treeNodeApi.on(node, eventName, listener);
 	},
-} as const;
+	clone<TSchema extends ImplicitFieldSchema>(
+		node: TreeFieldFromImplicitField<TSchema>,
+	): Unhydrated<TreeFieldFromImplicitField<TSchema>> {
+		// If it's a primitive type (i.e., not a tree node), return it as is.
+		if (!isTreeNode(node)) {
+			return node;
+		}
+
+		const kernel = getKernel(node);
+		// For unhydrated nodes, we can create a cursor by calling `cursorFromInsertable` because the node
+		// hasn't been inserted yet. We can then create a new node from the cursor.
+		if (!kernel.isHydrated()) {
+			return createFromCursor(
+				kernel.schema,
+				cursorFromInsertable(kernel.schema, node),
+			) as Unhydrated<TreeFieldFromImplicitField<TSchema>>;
+		}
+
+		// For hydrated nodes, we need to create a new cursor in the forest and then create a new node from
+		// the cursor.
+		const forest = (kernel.context as HydratedContext).flexContext.checkout.forest;
```

Comment:
It should be possible to avoid this cast with something like:
```suggestion
		if (!kernel.context.isHydrated()) {
			return createFromCursor(
				kernel.schema,
				cursorFromInsertable(kernel.schema, node),
			) as Unhydrated<TreeFieldFromImplicitField<TSchema>>;
		}

		// For hydrated nodes, we need to create a new cursor in the forest and then create a new node from
		// the cursor.
		const forest = kernel.context.flexContext.checkout.forest;
```

Reply:
I think that `FlexTreeContext` has the `isHydrated` typeguard, but `Context` currently does not, so we should just add it.

Reply:
I updated it to check for flexContext.isHydrated and that did the trick.